pac4j-core/src/main/java/org/pac4j/core/client/Clients.java

```diff
@@ -104,7 +105,10 @@ protected void internalInit() {
      */
     public Client findClient(final WebContext context) {
         final String name = context.getRequestParameter(this.clientNameParameter);
-        CommonHelper.assertNotBlank("name", name);
+		if (name == null && defaultClient != null) {
```

Comment:
What is this `defaultClient` for? It doesn't seem related to the resource changes...